Configure Tentacle on the server

taliesins opened the pull request:
When setting up the Tentacle on the client, we want to update the Tentacle settings on Octopus server. So Tentacle will be ready as a deployment target as soon as recipe has run.

If Tentacle changes certificate, re-running will configure Octopus server for new settings.

taliesins committed:
Register or update tentacle with Octopus server when thumbprint of the local machine changes
Install latest calamari if node does not have it
Support multiple roles and environments with the use of a comma separated list

taliesins committed:
Escape variables

taliesins committed:
Dot include

taliesins committed:
An exception is throw if we attempt to get a machine or an environment that does not exist. So catch the exception and assume $null value.

taliesins committed:
Remove accidentally included quote

taliesins committed:
Update tentacle uri if its different
Update thumbprint if its different

taliesins committed:
Need to update endpoint for changes to occur

taliesins committed:
Get-OctopusTask is needed when running Start-OctopusCalamariUpdate

taliesins committed:
Remove accidentally added quote

taliesins committed:
Read thumbprint from tentacle.config

taliesins committed:
Start a health check after reconfiguring a tentacle

taliesins committed:
Make it easier for editors to work out which chunks are related

taliesins committed:
Use default action

taliesins committed:
If internet explorer first run setup has not yet run Invoke-WebRequest will fail, but if you use basic parsing it works as expected

taliesins committed:
Merge branch 'master' of https://github.com/burnzy/octopus-cookbook

# Conflicts:
#	attributes/default.rb
#	recipes/create_environment.rb
#	recipes/register_tentacle.rb

spuder commented:
+1 looking for this functionality. 

The travis build shows `net-ssh` could not be installed. Is that the only problem with this merge request?

taliesins committed:
A hash is preferred over a comma separated string, as Chef can merge and override hash values.

So changed from:
          'octopus' => {
            'api' => {
                'key' => octopus_api_key
            },
            'tentacle' => {
              'role' => 'octopus,portal',
              'environment' => environment_number
            }
          }
to:
          'octopus' => {
            'api' => {
                'key' => octopus_api_key
            },
            'tentacle' => {
              'role' => {
                  'octopus' => {:enable => true},
                  'portal' => {:enable => true},
                  'portal-prototype' => {:enable => false}
              },
              'environment' => environment_number
            }
          }

Commit:
Add new recipes for creating octopus teams, adding machines to environments and fix get-machine powershell

vdex42 committed:
Update octoposh.ps1.erb

Commit:
Modify header, and add guard clause

taliesins committed:
Merge pull request #1 from vdex42/master

New recipes for creating octopus teams and  adding machines environments

Commit:
Fix environment array split issue caused by removal of invoke line
Fix guard clause try-catch always erroring

taliesins committed:
Merge pull request #2 from vdex42/master

Fix environment array split issue
